include/bluetooth/gatt.h

```diff
@@ -617,24 +617,36 @@ ssize_t bt_gatt_attr_write_ccc(struct bt_conn *conn,
 			       const struct bt_gatt_attr *attr, const void *buf,
 			       u16_t len, u16_t offset, u8_t flags);
 
-/** @def BT_GATT_CCC_MANAGED
- *  @brief Managed Client Characteristic Configuration Declaration Macro.
+
+/** @def BT_GATT_CCC_INITIALIZER
+ *  @brief Initialize Client Characteristic Configuration Declaration Macro.
  *
- *  Helper macro to declare a Managed CCC attribute.
+ *  Helper macro to initialize a Managed CCC attribute value.
  *
  *  @param _changed Configuration changed callback.
  *  @param _write Configuration write callback.
  *  @param _match Configuration match callback.
  */
-#define BT_GATT_CCC_MANAGED(_changed, _write, _match)			\
-	BT_GATT_ATTRIBUTE(BT_UUID_GATT_CCC,				\
-			BT_GATT_PERM_READ | BT_GATT_PERM_WRITE,		\
-			bt_gatt_attr_read_ccc, bt_gatt_attr_write_ccc,	\
-			(&(struct _bt_gatt_ccc) {			\
-				.cfg = {},				\
-				.cfg_changed = _changed,		\
-				.cfg_write = _write,			\
-				.cfg_match = _match }))
+#define BT_GATT_CCC_INITIALIZER(_changed, _write, _match) \
+	{                                            \
+		.cfg = {},                           \
```

Comment:
Do we actually need this line?

Reply:
Well I guess if the variable is not declared as static that would be needed.

Reply:
Only if it has no initialiser. Here it does have an initialiser (the stuff between `{ .. }`) and as per C-spec any elements not explicitly mentioned will be implicitly initialised to zero.

Reply:
To be clear: I'm talking about the parent struct, not the `.cfg` initialiser. If you omit mentioning .cfg then it'll be implicitly initialised to zero, regardless of the parent struct being static or not.

Reply:
Didn't know that, well in that case we can probably remove .cfg = {}, I was assuming it not to be the case since we do things like:

#define Z_TIMER_INITIALIZER(obj, expiry, stop) \
	{ \
	.timeout = { \
		.node = {},\

Reply:
@jhedberg can you please post a link to the initializer section of the spec? Now that we found compound literals we may as well build a list of tricks that are good to know.

Reply:
Actually, here it is:
https://gcc.gnu.org/onlinedocs/gcc/Designated-Inits.html

>Omitted fields are implicitly initialized the same as for objects that have static storage duration.